### DISCUSSION TITLE
Bokeh plots visualization issue (and one more)

### DISCUSSION THREAD

**Discussion by isergii** [datapane/datapane#119] [Help] [answered]
Hi Datapane team,

I'm struggling with a problem related to showing Bokeh plots in saved html file using datapane. Altair works just fine in the same conditions (didn't check other plotting libraries).

Three issues found (numbers here correspond to numbers on images below):

1. [Bokeh] Plots are duplicated and placed near each other (no such issue when Altair is used).
2. [Datapane] Two redundant "Select" components created (maybe I did smth wrong, not sure here)
3. [Bokeh] After clicking on some Select button, plots from other Select group disappear. Good news is that duplicate plots (issue 1) disappear as well. There is no such a problem if we switch to Altair - try to replace plot1() and plot(2) functions to plot3() and plot4() - check the code below.

**Before clicking "Plots block 2. Selection 2." button (just page loaded):**
![image](https://user-images.githubusercontent.com/8494593/120989389-2bd82f80-c788-11eb-845a-0fcdd41c9949.png)

**After clicking "Plots block 2. Selection 2." button:**
![image](https://user-images.githubusercontent.com/8494593/120989445-3c88a580-c788-11eb-8a61-dea1f7216fe4.png)

**Versions:**
Datapane: 0.11.7
Bokeh: 2.2.0

**Code to replicate:**
```
import numpy as np
import datapane as dp
from bokeh.transform import factor_cmap
from bokeh.plotting import figure
from bokeh.models import ColumnDataSource
from bokeh.palettes import Spectral6
from bokeh.sampledata.iris import flowers
import altair as alt
import pandas as pd


def plot1():
    fruits = ['Apples', 'Pears', 'Nectarines', 'Plums', 'Grapes', 'Strawberries']
    counts = [5, 3, 4, 2, 4, 6]
    source = ColumnDataSource(data=dict(fruits=fruits, counts=counts))
    p = figure(x_range=fruits, plot_height=350, toolbar_location=None, title="Fruit Counts")
    p.vbar(x='fruits', top='counts', width=0.9, source=source, legend_field="fruits",
           line_color='white', fill_color=factor_cmap('fruits', palette=Spectral6, factors=fruits))
    return p


def plot2():
    colormap = {'setosa': 'red', 'versicolor': 'green', 'virginica': 'blue'}
    colors = [colormap[x] for x in flowers['species']]
    p = figure(title="Iris Morphology")
    p.circle(flowers["petal_length"], flowers["petal_width"],
             color=colors, fill_alpha=0.2, size=10)
    return p


def plot3():
    source = pd.DataFrame({
        'a': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
        'b': [28, 55, 43, 91, 81, 53, 19, 87, 52]
    })

    return alt.Chart(source).mark_bar().encode(
        x='a',
        y='b'
    )


def plot4():
    x = np.arange(100)
    source = pd.DataFrame({
        'x': x,
        'f(x)': np.sin(x / 5)
    })

    return alt.Chart(source).mark_line().encode(
        x='x',
        y='f(x)'
    )


def make_report():
    report = [
        dp.Select(
            blocks=[
                dp.Group(blocks=[dp.Plot(plot1())], label='Plots block 1. Selection 1.'),
                dp.Group(blocks=[dp.Plot(plot1())], label='Plots block 1. Selection 2.'),
            ]
        ),
        dp.Select(
            blocks=[
                dp.Group(blocks=[dp.Plot(plot2())], label='Plots block 2. Selection 1.'),
                dp.Group(blocks=[dp.Plot(plot2())], label='Plots block 2. Selection 2.'),
            ]
        ),
    ]

    dp.Report(blocks=report, type=dp.ReportType.REPORT).save(
        path='datapane_report.html',
        open=True,
    )


if __name__ == "__main__":
    make_report()
```

**Comment by mands** [comment, marked as answer]
Hi @isergii ,

Thanks for raising and for the clear bug report and example code. We believe we've identified the issue and have released a fix in version 0.11.8 - could you check to see if it resolves the issue?

Thanks!

**Reply by isergii** [reply]
Hi @mands ,

Thanks for reply and the fix in new update. I tested new version in a different ways and didn't find issues (at least in the scope of what I reported). 

I will let you know if find something else. Thanks for a great product and I hope and looking for many cool stuff related to datapane!
